filestack/src/main/java/com/filestack/android/FsActivity.java

```diff
@@ -166,20 +166,24 @@ protected void onCreate(Bundle savedInstanceState) {
         nav.setItemIconTintList(ColorStateList.valueOf(theme.getAccentColor()));
 
         Config config = (Config) intent.getSerializableExtra(FsConstants.EXTRA_CONFIG);
-        String sessionToken = preferences.getString(PREF_SESSION_TOKEN, null);
-        Util.initializeClientIfNeeded(config, sessionToken);
-
-        if (savedInstanceState == null) {
-            Util.getSelectionSaver().clear();
-
-            selectedSource = sources.get(0);
-            nav.getMenu().performIdentifierAction(Util.getSourceIntId(selectedSource), 0);
-            if (drawer != null) {
-                drawer.openDrawer(GravityCompat.START);
+        if (config != null) {
+            String sessionToken = preferences.getString(PREF_SESSION_TOKEN, null);
+            Util.initializeClientIfNeeded(config, sessionToken);
+
+            if (savedInstanceState == null) {
+                Util.getSelectionSaver().clear();
+                selectedSource = sources.get(0);
+                nav.getMenu().performIdentifierAction(Util.getSourceIntId(selectedSource), 0);
+                if (drawer != null) {
+                    drawer.openDrawer(GravityCompat.START);
+                }
+            } else {
+                selectedSource = savedInstanceState.getString(STATE_SELECTED_SOURCE);
+                shouldCheckAuth = savedInstanceState.getBoolean(STATE_SHOULD_CHECK_AUTH);
             }
         } else {
-            selectedSource = savedInstanceState.getString(STATE_SELECTED_SOURCE);
-            shouldCheckAuth = savedInstanceState.getBoolean(STATE_SHOULD_CHECK_AUTH);
+            setResult(FsConstants.RESULT_FILE_STACK_ERROR);
```

Comment:
Are we going to handle this in the Fieldwire app?

Reply:
yup